services/src/functional-tests/src/test/groovy/org/openkilda/functionaltests/spec/flows/FlowCrudV2Spec.groovy

```diff
@@ -516,6 +520,115 @@ class FlowCrudV2Spec extends HealthCheckSpecification {
         Wrappers.wait(WAIT_OFFSET) { assert !islUtils.getIslInfo(newIsl).isPresent() }
     }
 
+    @Ignore("Encapsulation type is not implemented yet in APIv2")
+    @Unroll
+    @Tags(HARDWARE)
+    def "System allows to create/update encapsulation type for a flow\
+(#encapsulationCreate.toString() -> #encapsulationUpdate.toString())"() {
+        given: "Two active not neighboring switches"
+        def noviflowSwitchesIds = topology.activeSwitches.findAll { !it.centec && it.ofVersion == "OF_13" }*.dpId
+        def (Switch srcSwitch, Switch dstSwitch) = topology.activeTraffGens*.switchConnected.findAll {
+            noviflowSwitchesIds.contains(it.dpId)
+        }
+        assumeTrue("There should be at least two active traffgens connected to NoviFlow switches for test execution",
+                srcSwitch != null && dstSwitch != null)
+        when: "Create a flow with #encapsulationCreate.toString() encapsulation type"
+        def flow = flowHelperV2.randomFlow(srcSwitch, dstSwitch)
+        flow.encapsulationType = encapsulationCreate
+        flowHelperV2.addFlow(flow)
+
+        then: "Flow is created with the #encapsulationCreate.toString() encapsulation type"
+        def flowInfo = northbound.getFlow(flow.flowId)
+        flowInfo.encapsulationType == encapsulationCreate.toString().toLowerCase()
+
+        and: "Flow is valid"
+        northbound.validateFlow(flow.flowId).each { direction -> assert direction.asExpected }
+
+        and: "The flow allows traffic"
+        def traffExam = traffExamProvider.get()
+        def exam = new FlowTrafficExamBuilder(topology, traffExam).buildBidirectionalExam(flow, 0)
```

Comment:
I think we can run the traffic in parallel in both directions here. Find examples in other specs

Reply:
adding `withPool` is not enough. 'withPool' enables you to use `eachParallel` instead of `each`. Otherwise you are not running anything in parallel

Reply:
http://www.gpars.org/